frontend-react/src/components/Spinner.tsx

```diff
@@ -1,32 +1,41 @@
 import { Oval } from "react-loader-spinner";
+import { ReactNode } from "react";
 
 interface SpinnerProps {
     size?: "default" | "fullpage" | "insidebutton";
     display?: boolean; // allows easier dynamic show/hide during load
+    message?: ReactNode;
 }
 
-function Spinner({ size = "default", display = true }: SpinnerProps) {
+function Spinner({
+    size = "default",
+    display = true,
+    message = "",
+}: SpinnerProps) {
     // map prop to css className
     const sizeClassName = {
         default: "grid-container rs-spinner-default",
         fullpage: "grid-container rs-spinner-default rs-spinner-fullpage",
         insidebutton: "rs-spinner-default rs-spinner-tiny",
     }[size];
     return (
-        <div
-            hidden={!display}
-            className={sizeClassName}
-            data-testid="rs-spinner"
-        >
-            <Oval
-                height={60}
-                width={60}
-                color="#2378c3"
-                secondaryColor="#aacdec"
-                strokeWidth={8}
-                strokeWidthSecondary={8}
-                ariaLabel="loading-indicator"
-            />
+        <div>
+            <div
+                hidden={!display}
+                className={sizeClassName}
+                data-testid="rs-spinner"
+            >
+                <Oval
+                    height={60}
+                    width={60}
+                    color="#2378c3"
+                    secondaryColor="#aacdec"
+                    strokeWidth={8}
+                    strokeWidthSecondary={8}
+                    ariaLabel="loading-indicator"
+                />
+            </div>
+            <span className="text-center">{message}</span>
```

Comment:
This is optional so we should make it conditionally rendered:

```tsx
{ message && <span className="text-center">{message}</span> }
```

Reply:
Would also be worth adding a test for this?

Reply:
Done